### DISCUSSION TITLE
Hook to retrieve historical contract events

### DISCUSSION THREAD

**Discussion by JohnSmithyy** [wevm/wagmi#800] [Q&A]
Is there a hook to retrieve historical contract events? I noticed there's one for useContractEvent but just wondering if there's one for historical events.

Ideally I'd like to query a contracts events from a range of blocks and find an event.

**Comment by tmm** [comment]
Nope, there isn't a hook like this yet.

**Reply by JohnSmithyy** [reply]
Any plans in the future? Also how are people currently doing this?

**Comment by Ding-Fan** [comment]
I made a demo to get event logs

```ts
  const provider = useProvider()

  const theContract = useContract({
    addressOrName: process.env.NEXT_PUBLIC_CONTRACT_ADDRESS as string,
    contractInterface: abi.abi,
    signerOrProvider: provider
  })

  const iface = new ethers.utils.Interface(abi.abi)

  const showLogs = async () => {

    console.log('theContract', theContract)
    const haha = await theContract.filters.NewDrug(address)
    console.log('the provider', theContract.provider)
    const hehe = await theContract.queryFilter(haha)

    console.log('hehe', hehe)
    // console.log('contract interface', theContract.interface)

    if (iface && hehe) {
      console.log('haha topics', haha.topics)

      hehe.forEach((filterTopic: any) => {
        const result = iface.parseLog(filterTopic)
        // const result = iface.parseLog({ topics: haha.topics, data: '' })
        console.log('result', result)
      })
    }
  }
```

**Reply by JohnSmithyy** [reply]
This looks great! Are there limitations on how far you can go to get event logs in terms of blocks?

**Comment by dev-cz** [comment]
> I made a demo to get event logs
> 
> ```ts
>   const provider = useProvider()
> 
>   const theContract = useContract({
>     addressOrName: process.env.NEXT_PUBLIC_CONTRACT_ADDRESS as string,
>     contractInterface: abi.abi,
>     signerOrProvider: provider
>   })
> 
>   const iface = new ethers.utils.Interface(abi.abi)
> 
>   const showLogs = async () => {
> 
>     console.log('theContract', theContract)
>     const haha = await theContract.filters.NewDrug(address)
>     console.log('the provider', theContract.provider)
>     const hehe = await theContract.queryFilter(haha)
> 
>     console.log('hehe', hehe)
>     // console.log('contract interface', theContract.interface)
> 
>     if (iface && hehe) {
>       console.log('haha topics', haha.topics)
> 
>       hehe.forEach((filterTopic: any) => {
>         const result = iface.parseLog(filterTopic)
>         // const result = iface.parseLog({ topics: haha.topics, data: '' })
>         console.log('result', result)
>       })
>     }
>   }
> ```

This is awesome. Any examples of getting all transactions but filtered by the current connected account rather than by block?

Or maybe by event and connected account. So it only retrieves thos specific txs for the connected account.

**Comment by Ding-Fan** [comment]
@JohnSmithyy  @dev-cz 
I did some search, looks doable from the documents
https://docs.ethers.io/v5/concepts/events/#events--filters
https://docs.ethers.io/v5/api/providers/types/#types--events-and-logs

maybe we could create examples and add it to the wagmi document as "cookbook"
